### PR TITLE
improve instructions in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by DanGrayson**
See the discussion at #35920.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,32 +2,55 @@
...
-* `git commit` with message formatted `foo: <insert details>`
-* [open a pull request](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request) and fix any failing tests
+### To contribute a fix to the `foo` formula
+
+If you are already well versed in the use of `git`, then you can find the local
+copy of the `homebrew-core` repository in this directory
+(`Homebrew/Library/Taps/homebrew/homebrew-core/`), modify the formula there
+leaving the section `bottle do ... end` unchanged, and prepare a pull request
+as you usually do.  Before submitting your pull request, be sure to test it
+with these commands:
```
Comment (SMillerDev) [inline CONTRIBUTING.md]: Might be better to refer to the pull_request template here as those might change.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,32 +2,55 @@
...
+
+```
+brew uninstall --force foo
+brew install --build-from-source foo
+brew test foo
+brew audit --strict foo
+```
+
+After testing, if you would like to force the corresponding bottles to be
```
Comment (SMillerDev) [inline CONTRIBUTING.md]: Maybe rephrase this from "would like" "think it is needed"?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,32 +2,55 @@
...
+
+If you are not already well versed in the use of `git`, then you may learn
+about it from the introduction at
+https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request and then proceed as
+follows:
+
+* run `brew edit foo` and make edits
+* leave the section `bottle do ... end` unchanged
+* test your changes using the commands listed above
+* run `git commit` with message formatted `foo: <insert details>`
```
Comment (SMillerDev) [inline CONTRIBUTING.md]: without the colon: `foo <insert details>`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,32 +2,55 @@
...
+* open a pull request as described in the introduction linked to above, wait for the automated test results, and fix any failing tests
```
Comment (SMillerDev) [inline CONTRIBUTING.md]: Could you refer these people to the test commands too.

**Review by SMillerDev — CHANGES_REQUESTED** [review]
I think this is already better than before but I think there's still some improvements to be made.

File: CONTRIBUTING.md (hunk 1 above)
Comment (DanGrayson) [reply CONTRIBUTING.md]: I think it's better to have access to the test commands while making the changes, rather than seeing them at the last minute, as a pull request is underway.  And even if the pull request template changes, the commands listed here will not be bad ones.

File: CONTRIBUTING.md (hunk 4 above)
Comment (DanGrayson) [reply CONTRIBUTING.md]: I did, look 2 lines above.

**Comment by DanGrayson** [comment]
Thanks for the feedback, now take a look.

**Comment by SMillerDev** [comment]
Looks good to me.

**Comment by DanGrayson** [comment]
Thanks.  Perhaps you could mark your review so it shows the changes are now accepted.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -2,32 +2,55 @@
...
+
+If you are not already well versed in the use of `git`, then you may learn
+about it from the introduction at
+https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request and then proceed as
+follows:
+
+* run `brew edit foo` and make edits
+* leave the section `bottle do ... end` unchanged
+* test your changes using the commands listed above
+* run `git commit` with message formatted `foo <insert details>`
```
Comment (claui) [inline CONTRIBUTING.md]:
```suggestion
* run `git commit` with message formatted `foo: <insert details>`
```

**Review by claui — CHANGES_REQUESTED** [review]

**Review by SMillerDev — APPROVED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (SMillerDev) [reply CONTRIBUTING.md]: I just checked again and the bot does use it. Sorry about that, I guess I have to change my own habbit too then 😅.

**Comment by DanGrayson** [comment]
Okay, I'll put the colon back in, on Tuesday.

**Comment by claui** [comment]
Thanks @DanGrayson! Apologies for the confusion.

**Comment by SMillerDev** [comment]
@claui ready to merge?

**Review by claui — APPROVED** [review]

**Comment by claui** [comment]
@SMillerDev I made a change in order to clarify the commit messages. Mind reviewing my change one last time before we merge?

**Comment by SMillerDev** [comment]
Looks good to me.

**Comment by DanGrayson** [comment]
Fine with me.

**Comment by claui** [comment]
Thanks for your contribution @DanGrayson! 🍻
